src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -167,6 +167,12 @@ public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, V
         var linePosition = new LinePosition(
             request.Range.Start.Line,
             request.Range.Start.Character);
+
+        if (!sourceText.IsLinePositionValid(linePosition, _logger))
+        {
+            return null;
+        }
+
         var hostDocumentIndex = sourceText.Lines.GetPosition(linePosition);
```

Comment:
Looks like this is getting the absolute index anyway, so in light of my other comment, perhaps this hole bit, from line 167 to here, could be rewritten as:

```
if (!request.Range.Start.TryGetAbsoluteIndex(sourceText, out var hostDocumentIndex))
{
    return null;
}
```

Reply:
Agreed, other than seems that we should log error then, no?

Reply:
Reporting telemetry to track down an issue is a good idea, but isn't that what is already happening with the PRISM bugs? I don't _think_ there is any user impact of the exceptions here, unless they're watching the output window like a hawk.

Reply:
I guess in that case, do we want this fix or do we want to scrap it and look for the root issue? I finally got two heap dumps I'm digging into.

Reply:
Per our conversation added the check to at least handle one special case where LSP allows line number to be equal to line count if specified character is 0.